src/modules/RA/DataLayer/components/TableConfigField.js

```diff
@@ -0,0 +1,194 @@
+import React from 'react';
+
+import { useTranslate, useInput } from 'react-admin';
+
+import TextField from '@material-ui/core/TextField';
+import Typography from '@material-ui/core/Typography';
+
+import Switch from '@material-ui/core/Switch';
+
+import { SortableContainer, SortableElement, sortableHandle } from 'react-sortable-hoc';
+
+import { makeStyles } from '@material-ui/core/styles';
+import Table from '@material-ui/core/Table';
+import TableBody from '@material-ui/core/TableBody';
+import TableCell from '@material-ui/core/TableCell';
+import TableContainer from '@material-ui/core/TableContainer';
+import TableHead from '@material-ui/core/TableHead';
+import TableRow from '@material-ui/core/TableRow';
+import Paper from '@material-ui/core/Paper';
+
+const useStyles = makeStyles({
+  table: {
+    minWidth: 650,
+  },
+  wrapper: {
+    position: 'relative',
+    height: '400px',
+    overflowY: 'scroll',
+  },
+  row: {
+    zIndex: 10,
+    backgroundColor: 'white',
+    display: 'flex',
+    '& td': {
+      display: 'block',
+      flex: 1,
+    },
+  },
+  handle: {
+    position: 'relative',
+    top: '1px',
+    display: 'block',
+    width: '18px',
+    height: '11px',
+    opacity: '.25',
+    marginRight: '20px',
+    cursor: 'row-resize',
+    background: 'linear-gradient(180deg,#000,#000 20%,#fff 0,#fff 40%,#000 0,#000 60%,#fff 0,#fff 80%,#000 0,#000)',
+  },
+});
+
+
+const DragHandle = sortableHandle(() => {
+  const classes = useStyles();
+  return (<span className={classes.handle} />);
+});
+
+const FieldRow = SortableElement(({ field, onChange, exportEnabled }) => {
+  const translate = useTranslate();
+
+  const handleChangeTextField = property => e => {
+    const newValue = { ...field };
+    newValue[property] = e.target.value;
+    onChange(newValue);
+  };
```

Comment:
Shouldn't this be defined with `useCallback()` with `[field, onChange]` as memoisation argument?

Reply:
I can't do that as a new function is called, but see new code.